src/components/SecurityHeader.js

```diff
@@ -14,64 +14,48 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+'use strict';
+
 import NetInfo from '@react-native-community/netinfo';
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
-import Icon from 'react-native-vector-icons/MaterialIcons';
+import React, { useEffect, useState } from 'react';
+import { View } from 'react-native';
 import { withNavigation } from 'react-navigation';
 
 import colors from '../colors';
-import fonts from '../fonts';
-import TouchableItem from './TouchableItem';
-
-class SecurityHeader extends React.Component {
-	state = {
-		isConnected: false
-	};
-
-	componentDidMount() {
-		this.unsubscribe = NetInfo.addEventListener(state => {
-			this.setState({ isConnected: state.isConnected });
-		});
-	}
-
-	componentWillUnmount() {
-		this.unsubscribe();
-	}
-
-	render() {
-		const { isConnected } = this.state;
-
-		if (!isConnected) {
-			return null;
-		}
-
-		return (
-			<TouchableItem onPress={() => this.props.navigation.navigate('Security')}>
-				<View style={{ alignItems: 'center', flexDirection: 'row' }}>
-					<Icon style={styles.headerSecureIcon} name="security" />
-					<Text style={styles.headerTextRight}>Not Secure</Text>
-				</View>
-			</TouchableItem>
-		);
-	}
+import IdentitiesSwitch from '../components/IdentitiesSwitch';
+import ButtonIcon from './ButtonIcon';
+
+function SecurityHeader({ navigation }) {
+	const [isConnected, setIsConnected] = useState(false);
+
+	useEffect(() => {
+		const subscribe = () =>
+			NetInfo.addEventListener(state => {
+				setIsConnected(state.isConnected);
+			});
+		return subscribe();
```

Comment:
seems this is missing an unsubscribe() on component unmount?

Reply:
I noticed this, but it's "just" a typo imho. It should be named **un*subscribed, and this will be called when the component unmounts (because it's what the `useEffect` returns)

It was named `this.unsubscribe` previously

Reply:
It is kind of strange because this `subscribe` directly return an unsubscribe function. Probably name it as `unsubscribe`. 
sample usage from [NetInfo](https://github.com/react-native-community/react-native-netinfo)
```javascript
// Subscribe
const unsubscribe = NetInfo.addEventListener(state => {
  console.log("Connection type", state.type);
  console.log("Is connected?", state.isConnected);
});

// Unsubscribe
unsubscribe();
```

Reply:
This is how I'm used to see subscriptions. Polkadot-js api does the same. So to do things right we can rename it yes, but I think the logic is correct.

Reply:
yeah, let's just refactor to save "future us" some stress

Reply:
now it is even cleaner:
```javascript
	useEffect(
		() =>
			NetInfo.addEventListener(state => {
				setIsConnected(state.isConnected);
			}),
		[]
	);
```